File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        checked = set()
        longest = 0
        for num in nums:
            if num in checked:
                continue

            cur_cnt = 0
            cur_num = num
            while cur_num in nums:
                checked.add(cur_num)
                cur_num += 1
                cur_cnt += 1
            
            longest = max(longest, cur_cnt)
        
        return longest
            

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0, r = nums.size() - 1;
        int m;
        while (l < r) {
            m = (l + r) / 2;
            if (nums[m] < nums[r]) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        return nums[l];
    }
};
